src/Writers/HTMLWriter.jl

```diff
@@ -419,14 +442,25 @@ function render_search(ctx)
         ul["#search-results"]
     )
 
-    htmldoc = DOM.HTMLDocument(
-        html[:lang=>"en"](
-            head,
-            body(navmenu, article),
-            script[:src => relhref(src, ctx.search_index_js)],
-            script[:src => relhref(src, ctx.search_js)],
+    # search_index_js only set for lunr search
+    if search == "lunr"
+        htmldoc = DOM.HTMLDocument(
+            html[:lang=>"en"](
+                head,
+                body(navmenu, article),
+                script[:src => relhref(src, ctx.search_index_js)],
+                script[:src => relhref(src, ctx.search_js)],
+            )
         )
-    )
+    else
+        htmldoc = DOM.HTMLDocument(
+            html[:lang=>"en"](
+                head,
+                body(navmenu, article),
+                script[:src => "https://cdn.jsdelivr.net/algoliasearch/3/algoliasearchLite.min.js"],
+            )
+        )
+    end
```

Comment:
Either way is fine, but could slightly reduce code duplication by doing something like:

```julia
search_scripts = if search == "lunr"
    script[:src => relhref(src, ctx.search_index_js)], script[:src => relhref(src, ctx.search_js)]
else
    script[:src => "https://cdn.jsdelivr.net/algoliasearch/3/algoliasearchLite.min.js"],
end

htmldoc = DOM.HTMLDocument(
    html[:lang=>"en"](
        head,
        body(navmenu, article),
        search_scripts...,
    )
)
```